sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -124,10 +132,11 @@ class RapidsHostMemoryStore(
     }
 
     override protected def releaseResources(): Unit = {
-      if (isInternalPoolAllocated) {
-        assert(buffer.getAddress >= pool.getAddress)
-        assert(buffer.getAddress < pool.getAddress + pool.getLength)
-        addressAllocator.free(buffer.getAddress - pool.getAddress)
+      allocationMode match {
+        case Pooled =>
+          assert(buffer.getAddress >= pool.getAddress)
+          assert(buffer.getAddress < pool.getAddress + pool.getLength)
+          addressAllocator.free(buffer.getAddress - pool.getAddress)
       }
```

Comment:
This will throw a scala.MatchError exception when it doesn't find a match.  We need a `case _ =>` to fallthrough here.
```suggestion
        case _ =>
      }
```

Reply:
Done.